core/designsystem/src/main/java/com/google/samples/apps/nowinandroid/core/designsystem/component/DynamicAsyncImage.kt

```diff
@@ -34,11 +42,29 @@ fun DynamicAsyncImage(
     placeholder: Painter? = null,
 ) {
     val iconTint = LocalTintTheme.current.iconTint
-    AsyncImage(
-        placeholder = placeholder,
+    SubcomposeAsyncImage(
+        error = {
+            if (placeholder != null) {
+                Image(
+                    painter = placeholder,
+                    contentDescription = "placeholder image",
+                )
+            }
+        },
         model = imageUrl,
         contentDescription = contentDescription,
         colorFilter = if (iconTint != null) ColorFilter.tint(iconTint) else null,
         modifier = modifier,
+        loading = {
+            Box(
+                modifier = Modifier,
```

Comment:
Since this is the default value, it can be removed.

```suggestion
```

Reply:
Done already 👍